models/search/SearchActiveRecord.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+namespace app\models\search;
+
+
+use Yii;
+use yii\base\Exception;
+use yii\helpers\Inflector;
+
+abstract class SearchActiveRecord extends \yii\elasticsearch\ActiveRecord
+{
+    /**
+     * @var array language analyzers for elasticsearch based on
+     * https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-lang-analyzer.html
+     */
+    public static $languages = [
+        // lang => analyzer name
+        'en' => 'english',
+        'de' => 'german',
+        'es' => 'spanish',
+        'fr' => 'french',
+        'he' => 'standard', // 'hebrew' // currently not supported by elasticsearch
+        'id' => 'indonesian',
+        'it' => 'italian',
+        'ja' => 'standard', // 'japanese' // currently not supported by elasticsearch
+        'pl' => 'standard', // 'polish' // currently not supported by elasticsearch
+        'pt' => 'portuguese',
+        'pt-br' => 'portuguese',
+        'ro' => 'romanian',
+        'ru' => 'russian',
+        'sv' => 'swedish',
+        'uk' => 'russian', // 'ukrainian', // currently not supported by elasticsearch
```

Comment:
https://www.elastic.co/guide/en/elasticsearch/plugins/master/analysis-ukrainian.html

Reply:
this requires installing a plugin, the benefits for our use case are minimal, not going to add that now.